src/message/mod.rs

```diff
@@ -489,6 +493,69 @@ impl Message {
         self.format(&mut out);
         out
     }
+
+    #[cfg(feature = "dkim")]
+    /// Format body for signing
+    pub(crate) fn body_raw(&self) -> Vec<u8> {
+        let mut out = Vec::new();
+        match &self.body {
+            MessageBody::Mime(p) => p.format(&mut out),
+            MessageBody::Raw(r) => out.extend_from_slice(r),
+        };
+        out.extend_from_slice(b"\r\n");
+        out
+    }
+
+    #[cfg(feature = "dkim")]
```

Comment:
This line would be better placed at line 555

Reply:
Sure, done